spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/src/test/java/org/springframework/cloud/function/adapter/gcloud/integration/LocalServerTestSupport.java

```
@@ -0,0 +1,190 @@
+		assertThat(myClassPath).isNotNull();
+
+		List<String> command = new ArrayList<>();
+		command.addAll(Arrays.asList(javaCommand.toString(), "-classpath", myClassPath, Invoker.class.getName()));
+
+		ProcessBuilder processBuilder = new ProcessBuilder().command(command).redirectErrorStream(true);
+		Map<String, String> environment = new HashMap<>();
+		environment.put("PORT", String.valueOf(port));
+		environment.put("K_SERVICE", "test-function");
+		environment.put("FUNCTION_SIGNATURE_TYPE", signatureType);
+		environment.put("FUNCTION_TARGET", target);
+		environment.put("MAIN_CLASS", springApplicationMainClass.getCanonicalName());
+		if (function != null) {
+			environment.put("spring.cloud.function.definition", function);
+		}
+		processBuilder.environment().putAll(environment);
+		Process serverProcess = processBuilder.start();
+		CountDownLatch ready = new CountDownLatch(1);
+		StringBuilder output = new StringBuilder();
+		Future<?> outputMonitorResult = EXECUTOR
+				.submit(() -> monitorOutput(serverProcess.getInputStream(), ready, output));
+		boolean serverReady = ready.await(5, TimeUnit.SECONDS);
+		if (!serverReady) {
+			serverProcess.destroy();
+			throw new AssertionError("Server never became ready");
+		}
+		return new ServerProcess(serverProcess, outputMonitorResult, output, port);
+	}
+
+	private static void monitorOutput(InputStream processOutput, CountDownLatch ready, StringBuilder output) {
+		try (BufferedReader reader = new BufferedReader(new InputStreamReader(processOutput))) {
+			String line;
+			while ((line = reader.readLine()) != null) {
+				if (line.contains(SERVER_READY_STRING)) {
+					ready.countDown();
+				}
+				System.out.println(line);
+				synchronized (output) {
+					output.append(line).append('\n');
+				}
+				if (line.contains("WARNING")) {
+					throw new AssertionError("Found warning in server output:\n" + line);
+				}
+			}
+		}
+		catch (IOException e) {
```

Why do we catch the exception and then throw another one? Maybe just let it propagate naturally?

---

Converting from checked to unchecked.